contracts/Proofs.sol

```diff
@@ -19,27 +27,46 @@ abstract contract Proofs is Periods {
     _verifier = verifier;
   }
 
-  mapping(SlotId => uint256) private _slotStarts;
+  mapping(SlotId => uint256) private _slotStarts; // TODO: Should be smaller then uint256
   mapping(SlotId => uint256) private _probabilities;
-  mapping(SlotId => uint256) private _missed;
+  mapping(SlotId => uint256) private _missed; // TODO: Should be smaller then uint256
   mapping(SlotId => mapping(Period => bool)) private _received;
   mapping(SlotId => mapping(Period => bool)) private _missing;
 
   function slotState(SlotId id) public view virtual returns (SlotState);
 
+  /**
+   * @return Number of missed proofs since Slot was Filled
+   */
   function missingProofs(SlotId slotId) public view returns (uint256) {
     return _missed[slotId];
   }
 
+  /**
+   * @param slotId Slot's ID for which the proofs should be resetted
+   * @notice Resets the missing proofs counter to zero
+   */
   function _resetMissingProofs(SlotId slotId) internal {
     _missed[slotId] = 0;
   }
 
+  /**
+   * @param id Slot's ID for which the proofs should be started to require
+   * @param probability Integer which specifies the probability of how often the proofs will be required. Lower number means higher probability.
```

Comment:
I actually don’t know how to describe `proofProbability`. Only when the challenge 
 modulo the proofProbability is 0 is there a proof required. And since the challenge is random, it seems like the proofProbability doesn’t change the probability of a proof being required.

Reply:
A higher number in the modulo means there's a smaller percentage of the total set of possible entropy numbers that will result in zero. Think of the degenerate case of when you use a number in the modulo that's larger than half of the maximum number allowed for the entropy: you'll get exactly one number for which this is zero. Since you're drawing numbers at random, the chance you hit this is $1/n$, where $n$ is the maximum value for the entropy.

If you use something like $2$, on the other hand, half of the random numbers you get will result in zero, so you get a proof probability of $50\%$.

Reply:
Yeah, honestly that is something I am not such a fan of, because you can not pretty much specify any probability between 100% and 50% if I understand this correctly, right @gmega ?

Reply:
Yeah, you're missing a bunch of points, really, cause you're getting $\frac{1}{n}$ where $n$ is a natural number. So you get  $100\\%, 50\\%, 33.34\\%, 25\\%, \cdots$

I believe this is done like that cause it's easier to implement. As to whether or not this is a problem, I guess it depends on the period length. If you have short periods (e.g. block time of Ethereum), then maybe it's not a problem as a setting above $50\\%$ would probably never be used anyway.

Reply:
Btw, regardless, I think that description is inaccurate as $\text{probability}$ does not specify a probability, $1/\text{probability}$ does, and even then it's the probability that a proof will be required at a single discrete time interval, which I guess you're calling a `Period` in the code. 

I'd be tempted to say that this describes _the average number of `Period`s between proofs_, but since we do some [funky stuff with the number I don't understand that well](https://github.com/codex-storage/codex-research/blob/master/design/storage-proof-timing.md#pointer-downtime), I'm afraid this would also be misleading. But this is definitely not a probability.

Reply:
> Yeah, you're missing a bunch of points, really, cause you're getting  where  is a natural number. So you get...

That is the thing - it is not natural number but integer (`uint256`): https://github.com/codex-storage/codex-contracts-eth/blob/master/contracts/Requests.sol#L19. Otherwise, I would agree with you...

Reply:
Thanks for the explanation, makes much more sense now @gmega  🙏 

> perhaps more loosely the proof frequency
Yes, I agree it makes much more sense to call this proof frequency, where we'd expect, on average, a proof to be required every $n$ proofs

Reply:
> Not sure I follow - isn't uint* supposed to represent the set of non-negative integers (i.e., the set of natural numbers)?

Yeah, sorry - had a brain fart 😅 I thought you meant real numbers 🙈 But yeah re-reading your original comment, makes sense that the "upper range" (eq. 100%-50%) won't be really used...